orca-pipelinetemplate/src/test/groovy/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/model/DeploymentModelSpec.groovy

```diff
@@ -0,0 +1,227 @@
+    ])
+  }
+
+  def "should resolve a cluster"() {
+    given:
+    def models = [
+        new DeploymentModel(clusters: [spinBase: enhm([requires: [resources: [cpu: "1", ram: "4gb", disk: "20gb"]]])])
+    ]
+
+    def dmResolver = new DeploymentModelResolver(models)
+
+    when:
+    def cluster = dmResolver.getCluster("spinBase")
+
+    then:
+    cluster != null
+    cluster.requires.resources.cpu == "1"
+  }
+
+  def "should resolve a cluster with environment with account"() {
+    given:
+    def models = [
+        new DeploymentModel(accounts: [test: enhm(
+            name: "test",
+            cloudProvider: "aws",
+            regions: ["us-east-1": [
+                name: "us-east-1",
+                zones: ["us-east-1c", "us-east-1d"]
+            ]]
+        )]),
+        new DeploymentModel(environments: [staging: enhm(account: "test", stack: "staging")]),
+        new DeploymentModel(clusters: [spinBase: enhm([environment: "staging", requires: [resources: [cpu: "1", ram: "4gb", disk: "20gb"]]])])
+    ]
+
+    def dmResolver = new DeploymentModelResolver(models)
+
+    when:
+    def cluster = dmResolver.getCluster("spinBase")
+
+    then:
+    cluster != null
+    cluster.requires.resources.cpu == "1"
+    cluster.environment.stack == "staging"
+    cluster.environment.account.cloudProvider == "aws"
+  }
+
+  def "should allow inline extension"() {
+    def models = [
+        new DeploymentModel(accounts: [test: enhm(
+            name: "test",
+            cloudProvider: "aws",
+            regions: ["us-east-1": [
+                name: "us-east-1",
+                zones: ["us-east-1c", "us-east-1d"]
+            ]]
+        )]),
+        new DeploymentModel(environments: [staging: enhm(account: "test", stack: "staging")]),
+        new DeploymentModel(clusters: [spinBase: enhm([
+            environment: [extends: "staging", "detail": "canary"],
+            requires: [
+                resources: [
+                    cpu: "1",
+                    ram: "4gb",
+                    disk: "20gb"]]])])
+    ]
+
+    def dmResolver = new DeploymentModelResolver(models)
+
+    when:
+    def cluster = dmResolver.getCluster("spinBase")
+
+    then:
+    cluster != null
+    cluster.requires.resources.cpu == "1"
+    cluster.environment.stack == "staging"
+    cluster.environment.account.cloudProvider == "aws"
+    cluster.environment.detail == "canary"
+  }
+
+  def "should fail on unknown extends at resolve time"() {
+    given:
+    def models = [
+        new DeploymentModel(clusters: [spinBase: enhm([ extends: 'unknown' ])])
+    ]
+    def dmResolver = new DeploymentModelResolver(models)
+
+    when:
+    dmResolver.getCluster("spinBase")
+
+    then:
+    thrown(NoSuchElementException)
+  }
+
+  def "should read yaml filearooni"() {
+
+    given:
+    def mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT)
+    def models = new Yaml().loadAll(getClass().getResourceAsStream("/gate.yml")).collect {
```

Comment:
there is an open issue in that jackson yaml doesn't support a multi document file